### PR TITLE
Create a new `VIconButton`

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Related to #460 by @fcoveram

## Description

This PR adds a new `VIconButton` component using the updated styles in the [Figma mockups](https://www.figma.com/file/GIIQ4sDbaToCfFQyKMvzr8/Openverse-Design-Library?type=design&node-id=4658-12477&mode=design&t=3iXHyuTBcFAAWjLe-4)

The old styles for `VIconButton` were used mainly in the popover/modal close buttons, audio controls and the buttons within the search bar. They will gradually be updated to use the new `VIconButton`.

## Testing Instructions

Check out the new `VIconButton` story. It should have all of the styles from the Figma mockup. Old buttons (listed above) should also look the same.

Check the rendered Story: https://docs.openverse.org/_preview/2697/storybook/?path=/docs/components-viconbutton--default-story

## Checklist

- [ ] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [ ] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: frontend/src/components/VIconButton/meta/VIconButton.stories.mdx [hunk 1]
```diff
@@ -33,17 +38,20 @@ export const Template = (args) => ({
 export const sizesTemplate = (args) => ({
   template: `
     <div class="flex gap-x-2">
-      <div v-for="size in ['tiny', 'small', 'medium', 'large', 'extra-large']" :key="size" class="flex flex-col items-center p-2">
+      <div v-for="size in args.baseButtonSizes" :key="size" class="flex flex-col items-center p-2">
```
Comment (AetherUnbound) [inline frontend/src/components/VIconButton/meta/VIconButton.stories.mdx:41]: Nice!

**Review by AetherUnbound — APPROVED** [review]
LGTM! It seems like we're losing the `tiny` and `extra-large` sizes, is that intentional?

**Comment by zackkrida** [comment]
@obulat could you clarify the intention of this PR? It creates the new icon button, but doesn't actually implement it anywhere, right? Does that mean that the button styles _do_ need to change in the app (which will require updating visual regression tests) and that we will do that in subsequent PR(s)?

I'm mostly curious about the approach taken and if it would make more sense to create the new VIconButton but backwards-compatible, with the old sizes included (to be deprecated later), rather than maintain two versions for a (likely brief) period.

**Comment by obulat** [comment]
> LGTM! It seems like we're losing the `tiny` and `extra-large` sizes, is that intentional?

Yes! Those sizes were previously used for Audio controls. Now the designs have updated buttons with fewer sizes - that's why we are updating.

**Comment by obulat** [comment]
> @obulat could you clarify the intention of this PR? It creates the new icon button, but doesn't actually implement it anywhere, right? Does that mean that the button styles _do_ need to change in the app (which will require updating visual regression tests) and that we will do that in subsequent PR(s)?

The `VIconButton` component is currently used in three ways:
- close buttons in popover/modals - the external sources popover close button is the use case that prompted this PR
- `VSearchBarButton` - these button styles should not be changed, and they can be re-factored to use the new `VIconButton` because that will simplify the sizes and focus handling.
- Audio controls - #2602 updates them, but after the designs of the audio track are updated to match them.

The change in "close" buttons will cause changes in the visual regression snapshots.

> I'm mostly curious about the approach taken and if it would make more sense to create the new VIconButton but backwards-compatible, with the old sizes included (to be deprecated later), rather than maintain two versions for a (likely brief) period.

Current component (`VOldIconButton`) has a very complicated sizes system: we have to set both the icon and the button size, and they are coupled together, so there are too many sizes and they are still not enough to match all of the buttons we use.
After the update to `VSearchBarButton`, we can audit the close buttons used throughout the app in one go (updating the snapshots), and then `VOldIconButton` will be completely removed in the PR that updates the Audio control.

**Review by fcoveram — APPROVED** [review]
It looks great ✨ 🚀 

I noticed two prop variants I wasn't aware of, but do not block the work made: `filled-transparent` and `transparent-tx`. The first one is also an oxymoron 😅

**Pull request merged**